Water bridge 2nd Order implementation

talagayev opened the pull request:
Changes made:

- Addition of code for 2nd Order water bridges
- Implementation of input variable order

we would need to see if we make the order input then in the Fingerprint class and have a default, currently being 1
that the user can then input in the Fingerprint class.

As for the higher water bridge orders we would need to see how we do the recognition of interactions within the 
water-water analysis multiple times. Maybe through some looping like some step wise matching

cbouy commented:
I moved the code to a new module to make things clearer, it was getting quite daunting in the `fingerprint.py` file.

I also simplified the code a bit since there was a bit too much repetition in the metadata part to my liking.

Next is getting in a less "for-loop" approach to implementing each higher order and finding something more generalisable. Most likely using `networkx` to build the actual water network and then finding all shortest paths of length \<order> in the graph that go from \<ligand residue> to \<protein residue>. Let's see if I have time to do that before the new year

talagayev commented:
> I moved the code to a new module to make things clearer, it was getting quite daunting in the `fingerprint.py` file.
> 
> I also simplified the code a bit since there was a bit too much repetition in the metadata part to my liking.
> 
> Next is getting in a less "for-loop" approach to implementing each higher order and finding something more generalisable. Most likely using `networkx` to build the actual water network and then finding all shortest paths of length <order> in the graph that go from <ligand residue> to <protein residue>. Let's see if I have time to do that before the new year

Ah nice, true it gets a little bit overwhelming with the `metadata`.

Ah, somehow I didn't think about the `networkx` option. I would probably be on holidays till the new year, but would be
able to take a look from January 2nd to assist :)
